### PR TITLE
Allows the surgery body zone selector to use other UI themes

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request
Allows uses of the body zone selector for non-base TGUI themes (that look good, that is), doesn't add any to the asset cache, however, as this is just groundwork

## Why It's Good For The Game
The potential of things for looking better for other TGUI themes

## Changelog
Nothing player-facing

### CODE REVIEW

File: tgui/packages/tgui/components/BodyZoneSelector.tsx [hunk 1]
```diff
@@ -74,7 +75,7 @@ export class BodyZoneSelector extends Component<
 
   render() {
     const { hoverZone } = this.state;
-    const { scale = 3, selectedZone } = this.props;
+    const { scale = 3, selectedZone, theme = 'midnight' } = this.props;
```
Comment (Mothblocks) [inline tgui/packages/tgui/components/BodyZoneSelector.tsx:78]: If we're defaulting this, then change the type to `theme?: string` and remove the theme parameter from SurgeryInitiator

**Review by Mothblocks — APPROVED** [review]
The code is fine but I'm really not happy at the thought of these being generated for more themes. It's already pretty slow (I think 90ms?) just to generate the one because the blend calls are so expensive.

**Comment by Mothblocks** [comment]
I would want more consideration put into that when the time comes

**Comment by optimumtact** [comment]
you would be better off not allowing it at all

File: tgui/packages/tgui/components/BodyZoneSelector.tsx (hunk 1 above)
Comment (Zonespace27) [reply tgui/packages/tgui/components/BodyZoneSelector.tsx:78]: done

**Comment by Mothblocks** [comment]
Thinking on it now, if your eventual plan is just to mirror the theme the user has, then I don't think I would want this because the contrast on the tgui background is not going to be good.

What exactly is the use case you're hoping for @Zonespace27 ?

**Comment by Zonespace27** [comment]
> Thinking on it now, if your eventual plan is just to mirror the theme the user has, then I don't think I would want this because the contrast on the tgui background is not going to be good.

This isn't the plan
This PR is making the back-end for the body zone selector more flexible, so that you can easily add a different UI-theme body selector to match with a different-than-default TGUI theme, instead of the selector being forced to always be the midnight theme.

**Comment by Zonespace27** [comment]
> But for use in what feature?

For here, nothing, this is a change for a downstream PRed here because of its potential use and avoidance of downstream edits

**Comment by Mothblocks** [comment]
Good enough for me

**Pull request merged**